src/cmap/connection_pool.ts

```diff
@@ -446,13 +467,17 @@ function destroyConnection(pool: any, connection: any, reason: any) {
   process.nextTick(() => connection.destroy());
 }
 
-function processWaitQueue(pool: any) {
+function processWaitQueue(pool: ConnectionPool) {
   if (pool.closed) {
     return;
   }
 
   while (pool.waitQueueSize) {
     const waitQueueMember = pool[kWaitQueue].peekFront();
+    if (!waitQueueMember) {
```

Comment:
If we ever run into this case this would lead to an infinite loop, instead I think you want to `shift` the value off and `continue`